factor/grace/classes/factor.php

```diff
@@ -84,8 +84,8 @@ public function has_input() {
      *
      * {@inheritDoc}
      */
-    public function get_state() {
-        global $USER;
+    public function get_state($noredir = false) {
```

Comment:
I'm okay with params and defaults, but negative sounding booleans was never something I was fond about. Could this be changed to something like `redirect = true` instead?